nautobot_golden_config/api/views.py

```diff
@@ -124,12 +128,24 @@ class RemediationSettingViewSet(NautobotModelViewSet):  # pylint:disable=too-man
     filterset_class = filters.RemediationSettingFilterSet
 
 
-class ConfigPlanViewSet(NautobotModelViewSet):  # pylint:disable=too-many-ancestors
+class ConfigPlanViewSet(
+    NautobotAPIVersionMixin, NotesViewSetMixin, ModelViewSetMixin, ReadOnlyModelViewSet
+):  # pylint:disable=too-many-ancestors
     """API viewset for interacting with ConfigPlan objects."""
 
     queryset = models.ConfigPlan.objects.all()
     serializer_class = serializers.ConfigPlanSerializer
     filterset_class = filters.ConfigPlanFilterSet
 
-    # Disabling POST as these should only be created via Job.
```

Comment:
I had to bring over `get_serializer_context` from `nautobot.extras.api.views.CustomFieldModelViewSet` since it's not a mixin and the base class is `ModelViewSet`, which is what we need to get rid of to make this view ReadOnly